src/orion/algo/asha.py

```diff
@@ -214,27 +237,37 @@ class ASHABracket(HyperbandBracket):
 
     """
 
-    def sample(self, num):
+    def __init__(
+        self,
+        asha: ASHA,
+        budgets: list[BudgetTuple],
+        repetition_id: int,
+    ):
+        super().__init__(hyperband=asha, budgets=budgets, repetition_id=repetition_id)
+        self.asha: ASHA = asha
+
+    def sample(self, num: int) -> list[Trial]:
         """Sample a new trial with lowest fidelity"""
         should_have_n_trials = self.rungs[0]["n_trials"]
+        # BUG: Shouldn't this be `sample_from_bracket`?
```

Comment:
Wait, how could this not fail? Does it mean this is never called?

Reply:
Never called, indeed.

Reply:
Oh wow, thanks for noticing this dead limb! We should section it. :)

Reply:
Fixed in eaa2254453102689c230a4b5fb12952b9be77b20